x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/RollupActionIT.java

```diff
@@ -7,82 +7,259 @@
+                        "volume": {
+                            "type": "double",
+                            "time_series_metric": "gauge"
+                        }
+                    }
+                }
+            },
+            "data_stream": { }
+        }""";
 
     @Before
-    public void refreshIndex() {
+    public void refreshAbstractions() {
         index = "index-" + randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
         policy = "policy-" + randomAlphaOfLength(5);
         alias = "alias-" + randomAlphaOfLength(5);
-        logger.info("--> running [{}] with index [{}], alias [{}] and policy [{}]", getTestName(), index, alias, policy);
-    }
+        dataStream = "ds-" + randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
 
-    public void testRollupIndex() throws Exception {
-        createIndexWithSettings(
-            client(),
+        logger.info(
+            "--> running [{}] with index [{}], data stream [{}], alias [{}] and policy [{}]",
+            getTestName(),
             index,
+            dataStream,
             alias,
-            Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            policy
         );
-        index(client(), index, "_id", "timestamp", "2020-01-01T05:10:00Z", "volume", 11.0);
-        RollupActionConfig rollupConfig = new RollupActionConfig(DateHistogramInterval.DAY);
+    }
 
-        createNewSingletonPolicy(client(), policy, "cold", new RollupILMAction(rollupConfig, null));
+    private void createIndex(String index, String alias) throws IOException {
+        Settings.Builder settings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexSettings.MODE.getKey(), IndexMode.TIME_SERIES)
+            .putList(IndexMetadata.INDEX_ROUTING_PATH.getKey(), List.of("metricset"))
+            .put(IndexSettings.TIME_SERIES_START_TIME.getKey(), "2006-01-08T23:40:53.384Z")
+            .put(IndexSettings.TIME_SERIES_END_TIME.getKey(), "2106-01-08T23:40:53.384Z")
+            .put(LifecycleSettings.LIFECYCLE_NAME, policy);
+
+        XContentBuilder builder = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("properties")
+            .startObject("@timestamp")
+            .field("type", "date")
+            .endObject()
+            .startObject("metricset")
+            .field("type", "keyword")
+            .field("time_series_dimension", true)
+            .endObject()
+            .startObject("volume")
+            .field("type", "double")
+            .field("time_series_metric", "gauge")
+            .endObject()
+            .endObject()
+            .endObject();
+        String mapping = Strings.toString(builder);
+        createIndexWithSettings(client(), index, alias, settings, mapping);
+    }
+
+    public void testRollupIndex() throws Exception {
```

Comment:
Should these tests also perform a simple query to ensure that they can query the data stream without any problems after the rollup?

Reply:
I added an assertion that the `IndexMetadata.INDEX_ROLLUP_STATUS` index setting is `RollupTaskStatus.SUCCESS` at the end of the process: [7345a57](https://github.com/elastic/elasticsearch/pull/87269/commits/7345a57ff42b461904e0722171298f935fad92f7)

For queries on rollup indexes there is the [`RollupActionSingleNodeTests` class](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/v2/RollupActionSingleNodeTests.java) that validates the result of the rollup process.